src/routes/paste.rs

```diff
@@ -195,12 +195,21 @@ pub async fn insert(
         .ok_or_else(|| StatusCode::UNSUPPORTED_MEDIA_TYPE.into_response())?;
 
     if content_type == headers::ContentType::form_url_encoded() {
+        let is_https = headers.get("origin").is_some_and(|value| {
+            value
+                .to_str()
+                .ok()
+                .is_some_and(|val| val.starts_with("https://"))
+        });
```

Comment:
I hope this logic does not break anything.

Reply:
Hardened the check to match the `host` header.